### ISSUE TITLE
Why should extractor UDF have outputs for every input row?

### ISSUE THREAD

**Issue by zifeishan** [dennybritz/deepdive#117]
There sholud be cases that UDF do not have output for each row. Does system require output for each input row?

I am having `ERROR java.lang.NullPointerException` in several different places. Although they can be saved in other manners, I am not sure why it is necessary to require output for each input tuple.

**Comment by dennybritz** [comment]
I don't think this is where the error comes from. Unless someone changed that code, extractors aren't required to output a tuple for each input row. You can output nothing, or many tuples for each input row.
